src/foundation.cs

```diff
@@ -8752,7 +8753,7 @@ partial interface NSItemProvider : NSCopying {
 
 		[Mac (10,12)][Async]
 		[Export ("registerCloudKitShareWithPreparationHandler:")]
-		void RegisterCloudKitShare ([BlockCallback] Action<CloudKitRegistrationPreparationHandler> preparationHandler);
+		void RegisterCloudKitShare (CloudKitRegistrationPreparationAction preparationHandler);
```

Comment:
The linker won't be able to resolve the old method (from a binary) so it's a breaking change.
Also it's not clear why one would need the attribute and not the other ? _maybe I forgot_